docs/source/monitoring.md

```diff
@@ -1,76 +1,234 @@
-   ```
-
-   If you want to tweak the Prometheus properties, for example it's assigned memory,
-   you can override it by adding a command line argument like this: `--set prometheus.resources.limits.memory=4Gi`
-   or edit values file located at `examples/common/monitoring/values.yaml`.
-
-   The `yq` command prepares and formats custom Prometheus rules included in Scylla Monitoring and passes them as
-   additional values to `helm install` command.
-   It results in creating additional PrometheusRules: custom resources used to mount the rules into Prometheus.
-
-1. Install Service Monitors
-
-    ServiceMonitors are used by the Prometheus to discover applications exposing metrics.
-
-    ```
-    # Scylla Service Monitor
-    kubectl apply -f examples/common/monitoring/scylla-service-monitor.yaml
-
-    # Scylla Manager Service Monitor
-    kubectl apply -f examples/common/monitoring/scylla-manager-service-monitor.yaml
-    ```
-
-1. Install dashboards
-
-    Scylla Monitoring comes with pre generated dashboards suitable for multiple Scylla versions.
-    In this example we will use dashboards for Scylla 4.6, and Scylla Manager 3.0.
-    Amend directory path to generated dashboards to version suitable for your deployment.
-
-   Now the dashboards can be created like this:
-   ```
-   # Scylla dashboards
-   for f in scylla-monitoring/grafana/build/ver_4.6/*.json; do
-     kubectl -n scylla-monitoring create configmap scylla-dashboard-"$( basename "${f}" '.json' )" --from-file="${f}" --dry-run=client -o yaml | kubectl label -f- --dry-run=client -o yaml --local grafana_dashboard=1 | kubectl apply --server-side -f-
-   done
-
-   # Scylla Manager dashboards
-   for f in scylla-monitoring/grafana/build/manager_3.0/*.json; do
-     kubectl -n scylla-monitoring create configmap scylla-manager-dashboard-"$( basename "${f}" '.json' )" --from-file="${f}" --dry-run=client -o yaml | kubectl label -f- --dry-run=client -o yaml --local grafana_dashboard=1 | kubectl apply --server-side -f-
-   done
-    ```
-
-    Once Grafana sidecar picks up these dashboards they should be accessible in Grafana.
-
-To access Grafana locally, run:
-    ```
-    kubectl -n scylla-monitoring port-forward deployment.apps/monitoring-grafana 3000
-    ```
-
-   You can find it on `http://127.0.0.1:3000` and login with the credentials `admin`:`admin`.
\ No newline at end of file
+And then use a similar command to access Grafana again:
+```console
+$ curl --fail --retry-all-errors 10 --max-time 3 --retry-max-time 5 --cacert <( echo "${GRAFANA_SERVING_CERT}" ) --resolve "test-grafana.test.svc.cluster.local:3000:127.0.0.1" -IL "https://test-grafana.test.svc.cluster.local:3000" --user "${GRAFANA_USER}:${GRAFANA_PASSWORD}"
+
+HTTP/1.1 200 OK
+Cache-Control: no-cache
+Content-Type: text/html; charset=UTF-8
+Expires: -1
+Pragma: no-cache
+X-Content-Type-Options: nosniff
+X-Frame-Options: deny
+X-Xss-Protection: 1; mode=block
+Date: Fri, 07 Apr 2023 14:32:09 GMT
+```
```

Comment:
I've tried to simplify this and also some of the templates. Please verify the instructions.

### Accessing Grafana

For accessing Grafana service from outside the Kubernetes cluster we'd recommend using an Ingress but there are many other ways. What matters when using the Ingress is to direct your packets to the ingress controller service/pods and have the correct TLS SNI field set by the caller when reaching out the service, so it is routed properly, and it can successfully validate the grafana serving certificate. Obviously this is easier when you are using a real DNS domain that resolves to your Ingress controller IP address but most of the client/tools allow to set the SNI field manually.

#### Prerequisites

For accessing Grafana, you first need to collect the serving CA and the credentials.

```console
$ GRAFANA_SERVING_CERT="$( kubectl -n scylla get secret/example-grafana-serving-ca --template '{{ index .data "tls.crt" }}' | base64 -d )"
$ GRAFANA_USER="$( kubectl -n scylla get secret/example-grafana-admin-credentials --template '{{ index .data "username" }}' | base64 -d )"
$ GRAFANA_PASSWORD="$( kubectl -n scylla get secret/example-grafana-admin-credentials --template '{{ index .data "password" }}' | base64 -d )"
```

#### Connecting through Ingress using a resolvable domain

In case of production cluster someone has already set up the Ingress controller and appropriate DNS records. Often there is a generic wildcard record like `*.app.mydomain` already pointing to the ingress controller external IP. For custom service domains, these are usually a CNAME pointing the Ingress controller A record.

Note: The ScyllaDBMonitoring example creates an Ingress object with `test-grafana.test.svc.cluster.local` DNS domain - you need to adjust this to your domain. Here it is represented as `example-grafana.apps.mydomain`.

Note: To test a resolvable domain from your machine without creating DNS records, you can adjust `/etc/hosts` or similar.

```console
$ curl --fail -s -o /dev/null -w '%{http_code}' -L --cacert <( echo "${GRAFANA_SERVING_CERT}" ) "https://example-grafana.apps.mydomain" --user "${GRAFANA_USER}:${GRAFANA_PASSWORD}"
200
```

#### Connecting through Ingress using an unresolvable domain

To connect to an ingress without a resolvable domain you first need to find out your Ingress controller IP that can be resolved externally. Again there are more ways to do it, not limited to the few examples shown here.

Unless stated otherwise, we will assume your ingress is running on port 443.
```console
$ INGRESS_PORT=443
```

##### Variants
###### Ingress ExternalIP

When you are running in a real cluster there is usually a cloud LB or a bare metal alternative providing you with an externally reachable IP.

```console
$ INGRESS_IP="$( kubectl -n=haproxy-ingress get service/haproxy-ingress --template='{{ ( index .loadBalancer.ingress 0 ).ip }}' )"
```

###### Ingress NodePort

NodePort is a bit of a less convenient but it's available in dev clusters as well.

```console
# FIXME: the template needs to prefer ExternalIP, if set. 
$ INGRESS_IP="$( kubectl get nodes --template='{{ range ( index .items 0 ).status.addresses }}{{ if eq .type "InternalIP" }}{{ .address }}{{ end }}{{ end }}' )"
$ INGRESS_PORT="$( kubectl -n=haproxy-ingress get services/haproxy-ingress --template='{{ range .spec.ports }}{{ if eq .port 443 }}{{ .nodePort }}{{ end }}{{ end }}' )"
```

###### Ingress Port Forwarding

Port forwarding may be useful, if your nodes are not reachable through a public IP, nor your Ingress, or in general for development purposes. 

```console
$ kubectl -n=haproxy-ingress port-forward services/haproxy-ingress 5000:443 >/dev/null &
$ INGRESS_IP="localhost"
$ INGRESS_PORT="5000"
```

##### Connection

```console
$ curl --fail -s -o /dev/null -w '%{http_code}' -L --cacert <( echo "${GRAFANA_SERVING_CERT}" ) "https://test-grafana.test.svc.cluster.local"  --resolve "test-grafana.test.svc.cluster.local:${INGRESS_PORT}:${INGRESS_IP}" --user "${GRAFANA_USER}:${GRAFANA_PASSWORD}"
200
```

Reply:
I guess we had a different perspective on how this should look like but this looks good to me - I'll vet the commands and fix the typos and other errors.